Add new evergreen mobile product pages [fix #10182]

craigcook opened the pull request:
## Description
Adds new mobile product pages, replacing the old unified mobile page. We now have separate pages for Android and iOS, as well as a new page for Focus, and a new mobile browser comparison page.

The Focus page has a new native-style mobile banner for Focus, and you'll need to use an iOS or Android device to see it.

The old unified page is being kept temporarily as a fallback for the main mobile index page until sufficient locales can update.

## Issue / Bugzilla link
#10182 

## Testing
http://localhost:8000/firefox/browsers/mobile/
http://localhost:8000/firefox/browsers/mobile/android/
http://localhost:8000/firefox/browsers/mobile/ios/
http://localhost:8000/firefox/browsers/mobile/focus/
http://localhost:8000/firefox/browsers/mobile/compare/

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/366822640

alexgibson commented on bedrock/exp/urls.py:
```diff
@@ -16,7 +16,6 @@
     page('opt-out', 'exp/opt-out.html'),
     page('firefox', 'exp/firefox/index.html', ftl_files=['firefox/home']),
     url(r'^firefox/new/$', views.new, name='exp.firefox.new'),
-    page('firefox/mobile', 'exp/firefox/mobile.html', ftl_files=['firefox/mobile']),
```
Is it worth adding a redirect for this URL, incase there are external links to it?

alexgibson commented on bedrock/firefox/templates/firefox/browsers/mobile/compare.html:
```diff
@@ -0,0 +1,424 @@
+{% endblock %}
+
+{% block js %}
+  {{ js_bundle('firefox_compare_mobile') }}
```
The JS bundle here is loading the send-to-device form but not initialising it, so the form posts to a blank page.

alexgibson commented on bedrock/firefox/templates/firefox/browsers/mobile/includes/s2d-focus.html:
```diff
@@ -0,0 +1,24 @@
+  {{ send_to_device(include_title=False, message_set='fx-focus', class='vertical', id='s2d-primary') }}
+{% else %}
+  <p class="c-inline-form-body">{{ ftl('mobile-focus-scan-the-qr-code-to-get-started') }}</p>
+  {% if LANG == 'de' %}
+  <div class="qr-code-wrapper">
+    <img src="{{ static('img/firefox/browsers/mobile/qr-klar.png') }}" id="focus-qr" width="400" height="400" alt="{{ ftl('mobile-focus-scan-alt-text') }}">
+  </div>
+  {% else %}
```
Would this conditional ever be seen? Seems like s2d is available for DE?

alexgibson commented on bedrock/firefox/templates/firefox/faq.html:
```diff
@@ -62,10 +62,10 @@ <h2>{{ ftl('what-is-firefox') }}</h2>
-              url3=url('firefox.mac'),
-              url4=url('firefox.linux'),
-              url5=url('firefox.mobile.index')
+              url2=url('firefox.new.protocol.download_windows'),
```
This page is throwing a 500 error for me with `Reverse for 'firefox.new.protocol.download_windows' not found.`

alexgibson commented on l10n/en/firefox/browsers/mobile/compare.ftl:
```diff
@@ -0,0 +1,67 @@
+### URL: https://www-dev.allizom.org/firefox/browsers/mobile/compare/
+
+# HTML page title
+mobile-compare-5-of-the-best = 5 of the best mobile Browsers in direct comparison
```
Should `Browsers` here be lower case?
```suggestion
mobile-compare-5-of-the-best = 5 of the best mobile browsers in direct comparison
```

alexgibson commented on l10n/en/firefox/browsers/mobile/compare.ftl:
```diff
@@ -0,0 +1,67 @@
+mobile-compare-conclusion = Conclusion
+mobile-compare-based-on-the = Based on the criteria we outlined — privacy, features, and syncing — there’s really only one mobile browser that satisfies all three, and that’s { -brand-name-firefox }. While similar in the features and syncing categories, { -brand-name-firefox } stands apart in privacy, because we feel it’s the essential component that allows you to just enjoy the internet — exactl ...
+mobile-compare-our-recommendations-aside = Our recommendations aside, finding the browser that’s right for you will always depend on your individual needs and what you value most as you navigate online.
+
+
+
```
Nit: remove blank lines?

alexgibson commented on media/css/firefox/browsers/mobile.scss:
```diff
@@ -0,0 +1,233 @@
+
+//* -------------------------------------------------------------------------- */
+
+// not yet recognized as valid by style lint, so we abstract it
+// $mq-reduced-motion: 'screen and (prefers-reduced-motion: reduce)';
+
+// main .mzp-l-content {
+//     max-width: 400px;
+
+//     @media #{$mq-md} {
+//         max-width: $content-max;
+//     }
+// }
```
Is this needed still?

alexgibson commented on media/js/firefox/browsers/mobile.js:
```diff
@@ -0,0 +1,24 @@
+        var formPrimary = new Mozilla.SendToDevice('s2d-primary');
+        formPrimary.init();
+
+        // var formSecondary = new Mozilla.SendToDevice('s2d-secondary');
+        // formSecondary.init();
```
Is this needed?

alexgibson reviewed (changes requested):
I'll definitely need a second pass at this one I think, but I took a first run through and left some comments. All the pages look great!

The only common design issue I noticed was the main page headlines bleed over the coloured background gradients at mid-viewport sizes.

<img src="https://user-images.githubusercontent.com/400117/132720514-d4787707-7d9c-4bd3-9fa0-2d6a337c42d9.png" alt="Screenshot 2021-09-09 at 16-26-36 Firefox Browser Android" width="400">

craigcook commented:
Rebased and updated to address most of the comments.

But it seems the `fx-focus` message set no longer exists in Basket, so marking this as Do Not Merge until we can get that sorted. If we can dig up the old translated emails we can set up a new message set for Focus but if we have to start fresh with a new email we might have to think of something else. Maybe we could launch the Focus page with just the QR code.

craigcook commented:
The email folks are going to rework the email for Focus and rather than wait any longer we've opted to move ahead with just the QR code on the Focus page, and we can add the widget for Focus when it's ready.

alexgibson reviewed (approved):
r+ ☎️
